### DISCUSSION TITLE
Music on Android while using a reverse proxy

### DISCUSSION THREAD

**Discussion by ericfri** [lostb1t/replex#90] [General]
I don't think the issue is Replex, but only started after enabling it. It seems to be something with the reverse proxy as when I disable it and go back to the normal remote access method, things work as expected.

The issue I'm having seems to be specific to Android clients. I attempt to play music, and the majority of the time I just get a message saying "An error occurred while playing this item" and it skips to the next song until things stop. Usually, there will be a brief second or two of the song but will rarely continue playing past that.

The issue is not present while using Plexamp on any device, just the Plex app.

Any thoughts?

**Comment by lostb1t** [comment]
anythin in the logs?you can verify if its replex or your reverse oroxy causing it by routing your domain to plex instead of replex with your reverse proxy.

**Reply by lostb1t** [reply]
if you do find out replex is the cause then

enable RUST_LOG=info,replex=trace" for replex

enable debug an verbose logging for plex (enable through the llex web interface)

and send me the logs after reproducing it

**Reply by ericfri** [reply]
It's not Replex, it's something about the reverse proxy. I spent a few hours last night testing. Basically, no matter what I've tried on the proxy settings, the music does not play on Android. But it does play everywhere else.

I went through a number of reddit posts and example configs, adding headers and such to my proxy config, and while everything else works, the music doesn't. 

So far I've been unable to find anything helpful in the Plex logs, but I'm still looking.

**Reply by Zorro92** [reply]
What does your docker compose look like or otherwise configuration for replex?

**Reply by ericfri** [reply]
I'll grab that later. But I had a thought. Instead of using a reverse proxy I'm thinking I could just port forward to replex and keep remote access enabled which would should send everything through replex.

**Reply by Zorro92** [reply]
I'm using a reverse proxy myself and can access my music library perfectly fine

**Reply by ericfri** [reply]
On Android? My music works on seemingly every device but my Nvidia shield and Android phone using the Plex app. Plexamp works fine on my phone. The Plex app on my Mac works fine. Web works fine.

**Reply by Zorro92** [reply]
Yes works on my pixel 7 pro

**Reply by lostb1t** [reply]
> I'll grab that later. But I had a thought. Instead of using a reverse proxy I'm thinking I could just port forward to replex and keep remote access enabled which would should send everything through replex. 

that wont work as replex wont be able to read the communication. between plex and clients as its encrypted by SSL

**Reply by ericfri** [reply]
I returned to this (life got busy... I had a baby) and no matter what I try, I can't get music to play correctly on my Nvidia Shield when a reverse proxy is enabled.

I've tried multiple proxy options, including:
- Nginx Proxy Manager
- Caddy V2
- Cloudflare tunnels

I have disabled remote access and set my custom access URL. With all proxy options, TV and movies play perfectly on the Shield, but music does not. 
All media plays fine on my Mac, my parent's Samsung TV, and every other client I have available.

I have tried tethering my Shield to my phone and disconnecting the Ethernet with the same results. 

The Plex logs show `DEBUG - Failed to stream media, client probably disconnected after 32365596 bytes: 104 - Connection reset by peer` when things don't work. But occasionally music will play just fine. It's very frustrating.


Does anybody have _any_ thoughts?

**Reply by lostb1t** [reply]
Congratz on the new addition :) 

are you using replex in between? if not you might try the plex forums or reddit forums for more answers

**Reply by ericfri** [reply]
Thanks. I posted on [Reddit](https://www.reddit.com/r/PleX/comments/17oh87x/plex_behind_a_reverse_proxy_breaks_music_on_my/?) but haven't received any useful feedback.
